pubsub/envelope.go

```diff
@@ -35,17 +35,7 @@ type CloudEventsEnvelope struct {
 
 // NewCloudEventsEnvelope returns CloudEventsEnvelope from data or a new one when data content was not
 func NewCloudEventsEnvelope(id, source, eventType, subject string, data []byte) *CloudEventsEnvelope {
-	var ce CloudEventsEnvelope
-	err := jsoniter.Unmarshal(data, &ce)
-
-	if err == nil && ce.ID != "" && ce.SpecVersion != "" && ce.Type != "" && ce.Source != "" && ce.DataContentType != "" {
-		// data was already CloudEvent
-		// assuming the likelihood of other structures having all these
-		//CloudEvent-specific fields is very unlikely
-		return &ce
-	}
-
-	// ensure valid input parameters
+	// defaults
 	if id == "" {
```

Comment:
shouldn't lines 49 - 50 go inside the if statement in line 55?

Reply:
Used again in 95 so for readability reasons, the code checks everything up front. Does not change the execution logic as this value is only set if not provided up stream. Our current unit tests pass bunch of empty strings so not to change the tests, I'm validating on entry.

Reply:
ok